fpdf/line_break.py

```diff
@@ -0,0 +1,237 @@
+from collections import namedtuple
+
+SOFT_HYPHEN = "\u00ad"
+HYPHEN = "\u002d"
+SPACE = " "
+NEWLINE = "\n"
+
+
+class Fragment:
+    def __init__(self, style, underlined, characters=None):
+        self.characters = [] if characters is None else characters
+        self.style = style
+        self.underline = underlined
+
+    @classmethod
+    def from_string(cls, string, style, underlined):
+        return cls(style, underlined, list(string))
+
+    def add_character(self, character):
+        self.characters.append(character)
+
+    def trim(self, index):
+        self.characters = self.characters[:index]
+
+    @property
+    def string(self):
+        return "".join(self.characters)
+
+    def __eq__(self, other):
+        return (
+            self.characters == other.characters
+            and self.style == other.style
+            and self.underline == other.underline
+        )
+
+
+text_line = namedtuple(
```

Comment:
Would you mind using CamelCase for naming those `namedtuple` ?
There are classes after all, so `TextLine` / `SpaceHint` / `HyphenHint` would make more sense IMHO.